KIALI-2839 Rewrite getting started (#122)
kiali/kiali.io: content/gettingstarted/index.adoc

== Installing Kiali

> Kiali can be installed in an OpenShift or Kubernetes cluster environment. For OpenShift only, a minimal preparation is necessary.

Reviewer: line 46 says "on Openshift" and here it is "in Openshift"

Author: I'm not sure, but maybe `on OpenShift` and `in an OpenShift cluster environment` are both correct?
Maybe I can simplify here and just write `Kiali can be installed on OpenShift or Kubernetes.` wdyt?

=== OpenShift Preparation

If you are installing on OpenShift, you need to grant the `cluster-admin` role to the user that is installing Istio and Kiali. If OpenShift is installed locally on the machine you are using, the following command should log you in as user `system:admin` which has this `cluster-admin` role:

[source,bash]
----
oc login -u system:admin
----

=== Install Istio

Make sure your cluster environment has Istio installed.
For details on how to install Istio, see the link:https://istio.io/docs/setup/[Istio Setup docs].

=== Install the Kiali Operator

The Kiali Operator is an implementation of a link:https://coreos.com/operators/[Kubernetes Operator]. It provides a method of packaging, deploying, and managing Kiali.

To install the Kiali Operator (and, by default, the Kiali CR), run the install script via this command:

[source,bash]
----
bash <(curl -L https://git.io/getLatestKialiOperator)
----

> It will pull the latest version of Kiali Operator and Kiali itself, and install them with default options. If Kiali was already installed, you will be prompted about removing it first.

Reviewer 2: ```suggestion
This will pull the latest version of the Kiali Operator and Kiali itself, and install them using the default options. If Kiali was already installed, you will be prompted to remove the older version.
```

> icon:lightbulb[size=1x]{nbsp} The Kiali Operator install script can be configured by setting environment variables when running it. For example, if you do not want the install script to install a Kiali CR (and thus avoid having the Kiali Operator install Kiali itself), set the OPERATOR_INSTALL_KIALI environment variable to "false" and then run the script. Read the comments at the top of the link:https://github.com/kiali/kiali/blob/master/operator/deploy/deploy-kiali-operator.sh[Kiali Operator install script] for details on what custom configuration settings are available to you.

Reviewer 2: ```suggestion
icon:lightbulb[size=1x]{nbsp} You can configure the Kiali Operator install script by setting environment variables when you run it. For example, if you do not want the install script to install a Kiali custom resource definition (and thus avoid having the Kiali Operator install Kiali itself), set the OPERATOR_INSTALL_KIALI environment variable to "false" and then run the script. Read the comments at the top of the link:https://github.com/kiali/kiali/blob/master/operator/deploy/deploy-kiali-operator.sh[Kiali Operator install script] for details on what custom configuration settings are available to you.
```

Reviewer 2: Custom resource or custom resource definition?  (I'm still trying to get the different between the two straight in my head)

Author: It's custom resource (basically, a CR is kind of an "implementation" of a CRD). I'll take your suggestion and will change later

icon:bullhorn[size=1x]{nbsp} The install script requires `envsubst` installed and in your PATH; you can get it via the GNU `gettext` package.

During the installation, you will be prompted for the authentication strategy you want to use among `login`, `anonymous` and `openshift`.
See the link:#_login_options[Login Options] section below for more information.

=== Configure Kiali

The Kiali Operator runs inside the `kiali-operator` namespace and listens for a Kiali CR (custom resource) creation or changes. When it sees one created, it will install Kiali using the settings it finds in the CR object. To see all the different settings you can configure in a Kiali CR, see link:https://github.com/kiali/kiali/blob/master/operator/deploy/kiali/kiali_cr.yaml[this fully documented example Kiali CR yaml file].

To edit from a terminal, run:

[source,bash]
----
oc edit kiali kiali -n kiali-operator
----

Or to create / update using a local file (here `kiali.yml`):

[source,bash]
----

> oc apply -f kiali.yml -n kiali-operator

Reviewer: yml or yaml? there are several variations in this doc with this extension

Author: I've set it to yaml

== Installing Kiali

Kiali can be installed in an OpenShift or Kubernetes cluster environment. For OpenShift only, a minimal preparation is necessary.

=== OpenShift Preparation

If you are installing on OpenShift, you need to grant the `cluster-admin` role to the user that is installing Istio and Kiali. If OpenShift is installed locally on the machine you are using, the following command should log you in as user `system:admin` which has this `cluster-admin` role:

[source,bash]
----
oc login -u system:admin
----

> 

Reviewer 3: I haven't tried minikube for a while. Is this not needed anymore?

Author: I wasn't able to make it work with just "minikube addons enable ingress". While looking at the minikube doc for the nginx ingress, it seemed some other commands were necessary too - but did not work either for me. Personally I prefer to use kubectl port-forward.

So, I don't know, my take is that we can remove that kind of opinionated how-to because we don't check constantly if it continues to work with version x.y.z of minikube or kubernetes, and instead we just provide some links to external doc and/or assume the user already knows better than us how he or she wants to manage ingresses.

=== Install Istio

Make sure your cluster environment has Istio installed.
For details on how to install Istio, see the link:https://istio.io/docs/setup/[Istio Setup docs].

=== Install the Kiali Operator

The Kiali Operator is an implementation of a link:https://coreos.com/operators/[Kubernetes Operator]. It provides a method of packaging, deploying, and managing Kiali.

To install the Kiali Operator (and, by default, the Kiali CR), run the install script via this command:

[source,bash]
----
bash <(curl -L https://git.io/getLatestKialiOperator)
----

> It will pull the latest version of Kiali Operator and Kiali itself, and install them with default options. If Kiali was already installed, you will be prompted about removing it first.

Author: Note, this depends on not-merged-yet PR https://github.com/kiali/kiali/pull/1061

Reviewer 4: kiali/kiali#1061 has now been merged.

icon:bullhorn[size=1x]{nbsp} The install script requires `envsubst` installed and in your PATH; you can get it via the GNU `gettext` package.

During the installation, you will be prompted for the authentication strategy you want to use among `login`, `anonymous` and `openshift`.
See the link:#_login_options[Login Options] section below for more information.

=== Configure Kiali

The Kiali Operator runs inside the `kiali-operator` namespace and listens for the creation of a Kiali CR (custom resource) object. When it sees one created, it will install Kiali using the settings it finds in the CR object. To see all the different settings you can configure in a Kiali CR, see link:https://github.com/kiali/kiali/blob/master/operator/deploy/kiali/kiali_cr.yaml[this fully documented example Kiali CR yaml file].

> 

Reviewer 5: In addition to a created file I think it also responds to an edit?  Also. maybe mention that the pod will be restarted after the changes are applied?

Author: I've just rewritten as "... and listens for a Kiali CR (custom resource) creation or changes.", not saying too much to keep it short.

To edit from a terminal, run:

[source,bash]
----
oc edit kiali kiali -n kiali-operator
----

Or to create / update using a local file (here `kiali.yml`):

[source,bash]
----
oc apply -f kiali.yml -n kiali-operator
----

> === Open the UI

Reviewer 4: In this section, it doesn't tell the person how to log in (what credentials to use). To avoid frustration on the user's part, just put a quick blurb in here to tell them something like,

`The credentials you use on Kiali's login screen depend on the authentication strategy that was configured for Kiali. See "Login Options" for more details.`

Author: @jmazzitelli done
